tests/e2e/driver/driver.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package driver
+
+import (
+	"k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DynamicPVTestDriver represents an interface for a CSI driver that supports DynamicPV
+type DynamicPVTestDriver interface {
+	// GetDynamicProvisionStorageClass returns a StorageClass dynamic provision Persistent Volume.
+	GetDynamicProvisionStorageClass(parameters map[string]string, reclaimPolicy *v1.PersistentVolumeReclaimPolicy, bindingMode *storagev1.VolumeBindingMode, namespace string) *storagev1.StorageClass
+}
+
+func getStorageClass(
+	generateName string,
```

Comment:
Actually this is whats its called in the spec, I would prefer to keep it as is since its not the full _generated_ name:
```
		ObjectMeta: metav1.ObjectMeta{
			GenerateName: generateName,
		},
```